evoHike.Backend/Controllers/TrailsController.cs

```diff
@@ -0,0 +1,21 @@
+﻿using evoHike.Backend.Data;
+using evoHike.Backend.Models;
+using evoHike.Backend.Services;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace evoHike.Backend.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class TrailsController(ITrailService trailService) : ControllerBase
+    {
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Trail>>> GetTrails()
+        {
+            var trails = await trailService.GetAllTrailsAsync();
```

Comment:
Itt is hiba kezelést használnék:
```
try
    {
        var trails = await _trailService.GetAllTrailsAsync();
        return Ok(trails);
    }
    catch (Exception ex)
    {
        return BadRequest(ex.Message);
    }
```
Itt annyi a logika ha a service-ben lenne hiba itt elkapnánk és küldenénk tovább majd a user-nek